### ISSUE TITLE
How to handle Struct and Value?

### ISSUE THREAD

**Issue by fdeantoni** [danburkert/prost#245]
I have a message as follows:
```
syntax = "proto3";
import "google/protobuf/struct.proto";

message SomeMessage {
    google.protobuf.Value arbitrary_json = 1;
}
```

In Go you can then stuff arbitrary json data in the `arbitrary_json` field as follows:
```
import "github.com/golang/protobuf/ptypes/struct"
import "github.com/golang/protobuf/jsonpb"

func example(json string) (*SomeMessage, error) {
    msg := &SomeMessage{ArbitraryJson: &structpb.Value{}}
    jsm := jsonpb.Marshaler{}
    if err := jsm.Unmarshal(json, msg.ArbitraryJson); err != nil {
        return nil, err
    }
    return msg, nil
}
```
(copied from [here](https://groups.google.com/forum/#!topic/protobuf/ZWp_REiHlF4))

I understand serializing from JSON is not available for Prost, but is something similar like stuffing a `HashMap<String, Value>` into a `StructValue` possible? I took a look at [tests/src/well_known_types.rs](https://github.com/danburkert/prost/blob/master/tests/src/well_known_types.rs) but got no closer to a possible solution.

**Comment by fdeantoni** [comment]
I was able to answer my own question. Will it make sense to create convenience methods so creating values is easier? Something like the following:
```rust
#[test]
fn test_well_known_types_value() {
    let number: Value = Value::from(10.0);
    println!("Number: {:?}", number);
    let null: Value = Value::null();
    println!("Null: {:?}", null);
    let string: Value = Value::from(String::from("Hello"));
    println!("String: {:?}", string);
    let mut map: BTreeMap<String, Value> = BTreeMap::new();
    map.insert(String::from("number"), number);
    map.insert(String::from("null"), null);
    map.insert(String::from("string"), string);
    let pb_struct: Value = Value::from(map);
    println!("Struct: {:?}", pb_struct);
    let list = vec![pb_struct, Value::from(100.0)];
    let pb_list: Value = Value::from(list);
    println!("List: {:?}", pb_list);
}
```
If yes, I can create a pull request and work on it.

**Comment by fdeantoni** [comment]
Addressed in pull request #255

**Comment by fdeantoni** [comment]
Closed in favour of #277.
